Convert Mat image to INDArray, When trying to convert Mat image to INDArray it is returning me INDArray null

I have this code and I do not understand why my IDNarray image is returning me null when I try convert Mat in INDArray. 
I using the android sutdio 3.0.1.


```java
//************************* Digit classification *******************************************************************
        for (int i = 0; i < rects.size() ; i++) {
            Rect rect = rects.get(i);
            digit = inverted.submat(rect.y, rect.y + rect.height, rect.x, rect.x + rect.width);
            Imgproc.resize(digit, digit, new Size(28, 28));

                NativeImageLoader nativeImageLoader = new NativeImageLoader(digit.height(), digit.width(), digit.channels());//Use the nativeImageLoader to convert to numerical matrix
                INDArray image = nativeImageLoader.asMatrix(digit);//put image into INDArray

            System.out.println("carregar modelo matrixes  " + image);
 }
```
output: carregar modelo matrixes  NULL

replace e deeplearning4j-nn with deeplearning4j-core work, but when compiling, this error appears below in the line: MultiLayerNetwork model = ModelSerializer.restoreMultiLayerNetwork (inputStream);

here is the code:
``` 
try {
                    // Load the pretrained network.
                   InputStream inputStream = getResources().openRawResource(R.raw.classecharacter);
                   MultiLayerNetwork model = ModelSerializer.restoreMultiLayerNetwork(inputStream);

                     NativeImageLoader nativeImageLoader = new NativeImageLoader(28, 28, 1);//Use the nativeImageLoader to convert to numerical matrix
                     INDArray imageArray = nativeImageLoader.asMatrix(digito);//put image into INDArray

```


here is the error  
```
E/AndroidRuntime: FATAL EXCEPTION: main
                  Process: com.example.ailton.converttogray, PID: 26958
                  java.lang.IllegalStateException: Could not execute method for android:onClick
                      at android.support.v7.app.AppCompatViewInflater$DeclaredOnClickListener.onClick(AppCompatViewInflater.java:293)
                      at android.view.View.performClick(View.java:5242)
                      at android.widget.TextView.performClick(TextView.java:10571)
                      at android.view.View$PerformClick.run(View.java:21196)
                      at android.os.Handler.handleCallback(Handler.java:739)
                      at android.os.Handler.dispatchMessage(Handler.java:95)
                      at android.os.Looper.loop(Looper.java:145)
                      at android.app.ActivityThread.main(ActivityThread.java:6938)
                      at java.lang.reflect.Method.invoke(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:372)
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404)
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199)
                   Caused by: java.lang.reflect.InvocationTargetException
                      at java.lang.reflect.Method.invoke(Native Method)
                      at java.lang.reflect.Method.invoke(Method.java:372)
                      at android.support.v7.app.AppCompatViewInflater$DeclaredOnClickListener.onClick(AppCompatViewInflater.java:288)
                      at android.view.View.performClick(View.java:5242) 
                      at android.widget.TextView.performClick(TextView.java:10571) 
                      at android.view.View$PerformClick.run(View.java:21196) 
                      at android.os.Handler.handleCallback(Handler.java:739) 
                      at android.os.Handler.dispatchMessage(Handler.java:95) 
                      at android.os.Looper.loop(Looper.java:145) 
                      at android.app.ActivityThread.main(ActivityThread.java:6938) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at java.lang.reflect.Method.invoke(Method.java:372) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199) 
                   Caused by: java.lang.ClassCastException: org.bytedeco.javacpp.indexer.LongBufferIndexer cannot be cast to org.bytedeco.javacpp.indexer.LongRawIndexer
                      at org.nd4j.linalg.api.buffer.BaseDataBuffer.put(BaseDataBuffer.java:1219)
                      at org.nd4j.linalg.api.ndarray.BaseShapeInfoProvider.createShapeInformation(BaseShapeInfoProvider.java:87)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:59)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:44)
                      at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:172)
                      at org.nd4j.linalg.cpu.nativecpu.NDArray.<init>(NDArray.java:77)
                      at org.nd4j.linalg.cpu.nativecpu.CpuNDArrayFactory.create(CpuNDArrayFactory.java:205)
                      at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:3928)
                      at org.nd4j.linalg.factory.Nd4j.createArrayFromShapeBuffer(Nd4j.java:2397)
                      at org.nd4j.linalg.factory.Nd4j.read(Nd4j.java:2436)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:254)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:342)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:361)
                      at com.example.ailton.converttogray.MainActivity.convertGray(MainActivity.java:314)
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at java.lang.reflect.Method.invoke(Method.java:372) 
                      at android.support.v7.app.AppCompatViewInflater$DeclaredOnClickListener.onClick(AppCompatViewInflater.java:288) 
                      at android.view.View.performClick(View.java:5242) 
                      at android.widget.TextView.performClick(TextView.java:10571) 
                      at android.view.View$PerformClick.run(View.java:21196) 
                      at android.os.Handler.handleCallback(Handler.java:739) 
                      at android.os.Handler.dispatchMessage(Handler.java:95) 
                      at android.os.Looper.loop(Looper.java:145) 
                      at android.app.ActivityThread.main(ActivityThread.java:6938) 
                      at java.lang.reflect.Method.invoke(Native Method) 
                      at java.lang.reflect.Method.invoke(Method.java:372) 
                      at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:1404) 
                      at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1199) 
```

I'm using android studio 3.0.1, with 'deeplearning4j-core', version: '1.0.0-SNAPSHOT'

the error appears when loading the model in ModelSerializer.restoreMultiLayerNetwork (inputStream); I do not know what happens.

as I said in the comments but above, when I use the deeplearning4j-nn-
1.0.0-beta works fine, but the output matrix in the asMatrix function returns me all values as zero.

```
                   Caused by: java.lang.ClassCastException: org.bytedeco.javacpp.indexer.LongBufferIndexer cannot be cast to org.bytedeco.javacpp.indexer.LongRawIndexer
                      at org.nd4j.linalg.api.buffer.BaseDataBuffer.put(BaseDataBuffer.java:1219)
                      at org.nd4j.linalg.api.ndarray.BaseShapeInfoProvider.createShapeInformation(BaseShapeInfoProvider.java:87)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:59)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:44)
                      at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:172)
                      at org.nd4j.linalg.cpu.nativecpu.NDArray.<init>(NDArray.java:77)
                      at org.nd4j.linalg.cpu.nativecpu.CpuNDArrayFactory.create(CpuNDArrayFactory.java:205)
                      at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:3928)
                      at org.nd4j.linalg.factory.Nd4j.createArrayFromShapeBuffer(Nd4j.java:2397)
                      at org.nd4j.linalg.factory.Nd4j.read(Nd4j.java:2436)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:254)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:342)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:361)
                      at org.deeplearning4j.examples.image_recognition_demo.MainActivity$AsyncTaskRunner.doInBackground(MainActivity.java:73)
                      at org.deeplearning4j.examples.image_recognition_demo.MainActivity$AsyncTaskRunner.doInBackground(MainActivity.java:54)

``` 
pom.xml 

```
<project >

        <modelVersion>4.0.0</modelVersion>
        <groupId>org.deeplearning4j</groupId>
        <artifactId>snapshots</artifactId>
        <version>1.0.0-SNAPSHOT</version>
        <dependencies>
            <dependency>
                <groupId>org.nd4j</groupId>
                <artifactId>nd4j-native-platform</artifactId>
                <version>1.0.0-SNAPSHOT</version>
            </dependency>
            <dependency>
                <groupId>org.deeplearning4j</groupId>
                <artifactId>deeplearning4j-core</artifactId>
                <version>1.0.0-SNAPSHOT</version>
            </dependency>
        </dependencies>
        <repositories>
            <repository>
                <id>sonatype-nexus-snapshots</id>
                <url>https://oss.sonatype.org/content/repositories/snapshots</url>
                <releases>
                    <enabled>false</enabled>
                </releases>
                <snapshots>
                    <enabled>true</enabled>
                    <updatePolicy>always</updatePolicy>
                </snapshots>
            </repository>
        </repositories>
</project>
```

still the same error.

```
I/art: Explicit concurrent mark sweep GC freed 370(32KB) AllocSpace objects, 0(0B) LOS objects, 40% free, 13MB/23MB, paused 707us total 56.590ms
W/System.err: SLF4J: Failed to load class "org.slf4j.impl.StaticLoggerBinder".
W/System.err: SLF4J: Defaulting to no-operation (NOP) logger implementation
W/System.err: SLF4J: See http://www.slf4j.org/codes.html#StaticLoggerBinder for further details.
I/art: Rejecting re-init on previously-failed class java.lang.Class<org.nd4j.versioncheck.VersionCheck$1>
I/art: Rejecting re-init on previously-failed class java.lang.Class<org.nd4j.versioncheck.VersionCheck$1>
W/linker: libnd4jcpu.so: unused DT entry: type 0x6ffffffe arg 0x1ee6b0
W/linker: libnd4jcpu.so: unused DT entry: type 0x6fffffff arg 0x2
W/linker: libjnind4jcpu.so: unused DT entry: type 0x1d arg 0x9e3db
W/linker: libjnind4jcpu.so: unused DT entry: type 0x6ffffffe arg 0xd54c0
W/linker: libjnind4jcpu.so: unused DT entry: type 0x6fffffff arg 0x3
W/linker: libopenblas.so: unused DT entry: type 0x6ffffffe arg 0x121bc
W/linker: libopenblas.so: unused DT entry: type 0x6fffffff arg 0x2
W/linker: libjniopenblas_nolapack.so: unused DT entry: type 0xf arg 0x9f29
W/linker: libjniopenblas_nolapack.so: unused DT entry: type 0x6ffffffe arg 0xe77c
W/linker: libjniopenblas_nolapack.so: unused DT entry: type 0x6fffffff arg 0x2
E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
                  Process: org.deeplearning4j.examples.image_recognition_demo, PID: 2271
                  java.lang.RuntimeException: An error occured while executing doInBackground()
                      at android.os.AsyncTask$3.done(AsyncTask.java:304)
                      at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:355)
                      at java.util.concurrent.FutureTask.setException(FutureTask.java:222)
                      at java.util.concurrent.FutureTask.run(FutureTask.java:242)
                      at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:231)
                      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
                      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
                      at java.lang.Thread.run(Thread.java:818)
                   Caused by: java.lang.ClassCastException: org.bytedeco.javacpp.indexer.LongBufferIndexer cannot be cast to org.bytedeco.javacpp.indexer.LongRawIndexer
                      at org.nd4j.linalg.api.buffer.BaseDataBuffer.put(BaseDataBuffer.java:1219)
                      at org.nd4j.linalg.api.ndarray.BaseShapeInfoProvider.createShapeInformation(BaseShapeInfoProvider.java:87)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:59)
                      at org.nd4j.linalg.cpu.nativecpu.DirectShapeInfoProvider.createShapeInformation(DirectShapeInfoProvider.java:44)
                      at org.nd4j.linalg.api.ndarray.BaseNDArray.<init>(BaseNDArray.java:172)
                      at org.nd4j.linalg.cpu.nativecpu.NDArray.<init>(NDArray.java:77)
                      at org.nd4j.linalg.cpu.nativecpu.CpuNDArrayFactory.create(CpuNDArrayFactory.java:205)
                      at org.nd4j.linalg.factory.Nd4j.create(Nd4j.java:3928)
                      at org.nd4j.linalg.factory.Nd4j.createArrayFromShapeBuffer(Nd4j.java:2397)
                      at org.nd4j.linalg.factory.Nd4j.read(Nd4j.java:2436)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:254)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:342)
                      at org.deeplearning4j.util.ModelSerializer.restoreMultiLayerNetwork(ModelSerializer.java:361)
                      at org.deeplearning4j.examples.image_recognition_demo.MainActivity$AsyncTaskRunner.doInBackground(MainActivity.java:73)
                      at org.deeplearning4j.examples.image_recognition_demo.MainActivity$AsyncTaskRunner.doInBackground(MainActivity.java:54)
                      at android.os.AsyncTask$2.call(AsyncTask.java:292)
```

--- a/datavec/datavec-data/datavec-data-image/src/main/java/org/datavec/image/loader/NativeImageLoader.java
+++ b/datavec/datavec-data/datavec-data-image/src/main/java/org/datavec/image/loader/NativeImageLoader.java
@@ -143,6 +143,10 @@ public class NativeImageLoader extends BaseImageLoader {
         return ALLOWED_FORMATS;
     }
 
+    public INDArray asRowVector(String filename) throws IOException {
+        return asRowVector(new File(filename));
+    }
+
     /**
      * Convert a file to a row vector
      *
@@ -173,6 +177,10 @@ public class NativeImageLoader extends BaseImageLoader {
     }
 
     public INDArray asRowVector(Mat image) throws IOException {
+        return asMatrix(image).ravel();
+    }
+
+    public INDArray asRowVector(org.opencv.core.Mat image) throws IOException {
         return asMatrix(image).ravel();
     }
 
@@ -214,6 +222,9 @@ public class NativeImageLoader extends BaseImageLoader {
         return mat2;
     }
 
+    public INDArray asMatrix(String filename) throws IOException {
+        return asMatrix(new File(filename));
+    }
 
     @Override
     public INDArray asMatrix(File f) throws IOException {
@@ -290,6 +301,10 @@ public class NativeImageLoader extends BaseImageLoader {
             return bufferMat;
         }
 
+    }
+
+    public Image asImageMatrix(String filename) throws IOException {
+        return asImageMatrix(filename);
     }
 
     @Override
@@ -491,6 +506,10 @@ public class NativeImageLoader extends BaseImageLoader {
         image.deallocate();
     }
 
+    public void asMatrixView(String filename, INDArray view) throws IOException {
+        asMatrixView(new File(filename), view);
+    }
+
     public void asMatrixView(File f, INDArray view) throws IOException {
         try (BufferedInputStream bis = new BufferedInputStream(new FileInputStream(f))) {
             asMatrixView(bis, view);
@@ -501,14 +520,29 @@ public class NativeImageLoader extends BaseImageLoader {
         transformImage(image, view);
     }
 
+    public void asMatrixView(org.opencv.core.Mat image, INDArray view) throws IOException {
+        transformImage(image, view);
+    }
+
     public INDArray asMatrix(Frame image) throws IOException {
         return asMatrix(converter.convert(image));
+    }
+
+    public INDArray asMatrix(org.opencv.core.Mat image) throws IOException {
+        INDArray ret = transformImage(image, null);
+
+        return ret.reshape(ArrayUtil.combine(new long[] {1}, ret.shape()));
     }
 
     public INDArray asMatrix(Mat image) throws IOException {
         INDArray ret = transformImage(image, null);
 
         return ret.reshape(ArrayUtil.combine(new long[] {1}, ret.shape()));
+    }
+
+    protected INDArray transformImage(org.opencv.core.Mat image, INDArray ret) throws IOException {
+        Frame f = converter.convert(image);
+        return transformImage(converter.convert(f), ret);
     }
 
     protected INDArray transformImage(Mat image, INDArray ret) throws IOException {
@@ -628,6 +662,10 @@ public class NativeImageLoader extends BaseImageLoader {
         return scaled;
     }
 
+
+    public ImageWritable asWritable(String filename) throws IOException {
+        return asWritable(new File(filename));
+    }
 
     /**
      * Convert a file to a INDArray

--- a/datavec/datavec-data/datavec-data-image/src/test/java/org/datavec/image/loader/TestNativeImageLoader.java
+++ b/datavec/datavec-data/datavec-data-image/src/test/java/org/datavec/image/loader/TestNativeImageLoader.java
@@ -17,6 +17,7 @@
 package org.datavec.image.loader;
 
 import org.apache.commons.io.IOUtils;
+import org.bytedeco.javacpp.Loader;
 import org.bytedeco.javacpp.indexer.UByteIndexer;
 import org.bytedeco.javacv.Frame;
 import org.bytedeco.javacv.Java2DFrameConverter;
@@ -110,7 +111,7 @@ public class TestNativeImageLoader {
         NativeImageLoader loader5 = new NativeImageLoader(h4, w4, ch4, NativeImageLoader.MultiPageMode.FIRST);
         INDArray array6 = null;
         try {
-            array6 = loader5.asMatrix(new ClassPathResource(path2MitosisFile).getFile());
+            array6 = loader5.asMatrix(new ClassPathResource(path2MitosisFile).getFile().getAbsolutePath());
         } catch (IOException e) {
             e.printStackTrace();
             fail();
@@ -141,7 +142,7 @@ public class TestNativeImageLoader {
         NativeImageLoader loader7 = new NativeImageLoader(h4, w4, ch6, NativeImageLoader.MultiPageMode.MINIBATCH);
         INDArray array8 = null;
         try {
-            array8 = loader7.asMatrix(new ClassPathResource(path2MitosisFile).getFile());
+            array8 = loader7.asMatrix(new ClassPathResource(path2MitosisFile).getFile().getAbsolutePath());
         } catch (IOException e) {
             e.printStackTrace();
         }
@@ -154,7 +155,7 @@ public class TestNativeImageLoader {
 
     @Test
     public void testAsRowVector() throws Exception {
-        BufferedImage img1 = makeRandomBufferedImage(0, 0, 1);
+        org.opencv.core.Mat img1 = makeRandomOrgOpenCvCoreMatImage(0, 0, 1);
         Mat img2 = makeRandomImage(0, 0, 3);
 
         int w1 = 35, h1 = 79, ch1 = 3;
@@ -344,6 +345,15 @@ public class TestNativeImageLoader {
         return c2.convert(c.convert(img));
     }
 
+    org.opencv.core.Mat makeRandomOrgOpenCvCoreMatImage(int height, int width, int channels) {
+        Mat img = makeRandomImage(height, width, channels);
+
+        Loader.load(org.bytedeco.javacpp.opencv_java.class);
+        OpenCVFrameConverter.ToOrgOpenCvCoreMat c = new OpenCVFrameConverter.ToOrgOpenCvCoreMat();
+
+        return c.convert(c.convert(img));
+    }
+
     Mat makeRandomImage(int height, int width, int channels) {
         if (height <= 0) {
             height = rng.nextInt() % 100 + 100;
@@ -366,7 +376,7 @@ public class TestNativeImageLoader {
 
     @Test
     public void testAsWritable() throws Exception {
-        File f0 = new ClassPathResource("datavec-data-image/testimages/class0/0.jpg").getFile();
+        String f0 = new ClassPathResource("datavec-data-image/testimages/class0/0.jpg").getFile().getAbsolutePath();
 
         NativeImageLoader imageLoader = new NativeImageLoader();
         ImageWritable img = imageLoader.asWritable(f0);
@@ -398,7 +408,7 @@ public class TestNativeImageLoader {
         m.setAccessible(true);
 
         File f1 = new ClassPathResource("datavec-data-image/voc/2007/JPEGImages/000005.jpg").getFile();
-        File f2 = new ClassPathResource("datavec-data-image/voc/2007/JPEGImages/000007.jpg").getFile();
+        String f2 = new ClassPathResource("datavec-data-image/voc/2007/JPEGImages/000007.jpg").getFile().getAbsolutePath();
 
         //Start with a large buffer
         byte[] buffer = new byte[20*1024*1024];

--- a/pom.xml
+++ b/pom.xml
@@ -293,7 +293,7 @@
 
         <javacpp.version>1.4.3-SNAPSHOT</javacpp.version>
         <javacpp-presets.version>1.4.2</javacpp-presets.version>
-        <javacv.version>1.4.2</javacv.version>
+        <javacv.version>1.4.3-SNAPSHOT</javacv.version>
         <openblas.version>0.3.0</openblas.version>
         <mkl.version>2018.3</mkl.version>
         <mkl-dnn.version>0.16</mkl-dnn.version>

